language/move-prover/bytecode/src/borrow_analysis.rs

```diff
@@ -260,15 +260,14 @@ impl BorrowInfo {
     ) {
         for (dest, edge) in outgoing.iter() {
             let mut path = prefix.to_owned();
-            path.extend(edge.flatten().into_iter().cloned());
+            path.extend(edge.clone().reverse().flatten().into_iter().cloned());
             if let Some(succs) = ret_info.borrows_from.get(dest) {
                 self.construct_hyper_edges(leaf, ret_info, path, succs);
             } else {
                 // Reached a leaf.
                 let edge = if path.len() == 1 {
                     path.pop().unwrap()
                 } else {
-                    path.reverse();
```

Comment:
Sorry for an unexpected chiming-in here:

I think the `reverse()` here is needed. The issue/bug is that `flatten()` is called too early.

Think about the case where we have
`path = [ Hyper(Index, Field) ]`

Doing a flatten-then-reverse yield
```
[ Hyper(Index, Field) ] --flatten--> [ Index, Field ] --reverse--> [ Field, Index ] --repack--> [ Hyper(Field, Index) ]
```
This will distort the original hyper-edge.

While doing a reverse-then-flatten yield
```
[ Hyper(Index, Field) ] --reverse--> [ Hyper(Index, Field) ] --flatten--> [ Index, Field ] --repack--> Hyper(Index, Field)
```
This preserves the original hyper-edge.

Reply:
The PR is here: #298